crates/wasmtime/src/externals.rs

```diff
@@ -864,6 +902,38 @@ impl Memory {
         MemoryType::from_wasmtime_memory(&self.wasmtime_export.memory.memory)
     }
 
+    /// Copies memory contents at the given offset into a buffer.
+    ///
+    /// Returns an error if the copy failed.
```

Comment:
Could this documentation detail the error cases as well? It also might be good to mention that the size of the copy is the size of the buffer itself.

Reply:
I expanded the docs for both `read` and `write` a bit.